webidl.txt

```diff
@@ -72,50 +46,104 @@ dictionary AudioDecoderParams {
   MediaTime? codecDelay;
 } 
 
-[Constructor()]
-interface AudioTrackWriter {
-  readonly attribute WritableStream writable; // of DecodedAudioPacket
-  readonly attribute MediaStreamTrack track;
+dictionary AudioDecoderInput {
+  Uint8Array data;
+  MediaTime timestamp;
 }
 
+dictionary AudioDecoderOutput {
+  AudioBuffer buffer;
+  // TODO: decode stats.
+}
 
-[Constructor(MediaStreamTrack track)]
-interface VideoTrackReader {
-  readonly attribute ReadableStream readable; // of DecodedVideoFrame
+dictionary AudioEncoderStaticSettings {
+  // probably opus-specific
+  bool fec = false;  // enabled or not
+  bool dtx = false;  // enabled or not
+  bool cbr = false;  // cbr or not (vbr if not)
+  bool speechMode = false;  // speech-specific mode or not
+}
+
+dictionary AudioEncoderDynamicSettings {
+  // not supported by all codecs
+  // null/unset means use the codec default
+  unsigned long? bitsPerSecond;  
+
+  // codec-specific
+  // null/unset means use the codec default
+  unsigned long? complexity;
 }
 
-interface DecodedVideoFrame {
+dictionary AudioEncoderSettings : AudioEncoderStaticSettings, AudioEncoderDynamicSettings {
+}
+
+dictionary AudioEncoderInit : AudioCodecParameters {
+  AudioEncoderSettings? settings;
+}
+
+dictionary AudioEncoderFrameSettings {
```

Comment:
Since we went with "AudioPacket", I guess this should be "PacketSettings".

Reply:
Actually removed this since I'm not sure it's useful to have frame-affiliated settings like it is for video.